types/src/addressable_entity.rs

```diff
@@ -647,26 +668,780 @@ impl KeyValueJsonSchema for EntryPointLabels {
     const JSON_SCHEMA_KV_NAME: Option<&'static str> = Some("NamedEntryPoint");
 }
 
+#[allow(missing_docs)]
+#[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+#[repr(u8)]
+pub enum EntityKindTag {
+    System = 0,
+    Account = 1,
+    SmartContract = 2,
+}
+
+impl TryFrom<u8> for EntityKindTag {
+    type Error = bytesrepr::Error;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(EntityKindTag::System),
+            1 => Ok(EntityKindTag::Account),
+            2 => Ok(EntityKindTag::SmartContract),
+            _ => Err(bytesrepr::Error::Formatting),
+        }
+    }
+}
+
+impl ToBytes for EntityKindTag {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        (*self as u8).to_bytes()
+    }
+
+    fn serialized_length(&self) -> usize {
+        U8_SERIALIZED_LENGTH
+    }
+
+    fn write_bytes(&self, writer: &mut Vec<u8>) -> Result<(), bytesrepr::Error> {
+        (*self as u8).write_bytes(writer)
+    }
+}
+
+impl FromBytes for EntityKindTag {
+    fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
+        let (package_kind_tag, remainder) = u8::from_bytes(bytes)?;
+        match package_kind_tag {
+            package_kind_tag if package_kind_tag == EntityKindTag::System as u8 => {
+                Ok((EntityKindTag::System, remainder))
+            }
+            package_kind_tag if package_kind_tag == EntityKindTag::Account as u8 => {
+                Ok((EntityKindTag::Account, remainder))
+            }
+            package_kind_tag if package_kind_tag == EntityKindTag::SmartContract as u8 => {
+                Ok((EntityKindTag::SmartContract, remainder))
+            }
+            _ => Err(bytesrepr::Error::Formatting),
+        }
```

Comment:
nit: we can use the `TryFrom` impl above:
```suggestion
        Ok((package_kind_tag.try_into()?, remainder))
```

Reply:
Used suggestion